### PR TITLE
Implement 'addTopics' Functionality in consumerGroup Module

### PR DESCRIPTION
**Pull request by LeiSun-1101**
This is related to https://github.com/SOHU-Co/kafka-node/issues/632

### CODE REVIEW

**Comment by LeiSun-1101** [comment]
The automatic tests fails sometimes for no reason. And each time different tests fail. I try several times and pass the tests.

File: lib/consumerGroup.js [hunk 1]
```diff
@@ -611,6 +611,48 @@ ConsumerGroup.prototype.sendOffsetCommitRequest = function (commits, cb) {
   }
 };
 
+ConsumerGroup.prototype.addTopics = function (topics, cb) {
+  topics = Array.isArray(topics) ? topics : [topics];
+  let self = this;
+
+  if (!this.client.ready) {
+    this.client.once('ready', this.addTopics(topics, cb));
```
Comment (hyperlink) [inline lib/consumerGroup.js]: should wrap the addTopics call in a fat arrow function.

File: test/test.consumerGroup.js [hunk 2]
```diff
@@ -1050,4 +1050,49 @@ describe('ConsumerGroup', function () {
...
   });
+
+  describe('#addTopics', function () {
+    let topic, newTopic, testMessage, consumerGroup;
+
+    before('create the topics', done => {
+      topic = uuid.v4();
+      newTopic = uuid.v4();
+      testMessage = uuid.v4();
+      consumerGroup = new ConsumerGroup({
```
Comment (hyperlink) [inline test/test.consumerGroup.js]: can we close the consumerGroup in the `after` ?

File: lib/consumerGroup.js (hunk 1 above)
Comment (LeiSun-1101) [reply lib/consumerGroup.js]: done()

File: lib/consumerGroup.js [hunk 3]
```diff
@@ -611,6 +611,48 @@ ConsumerGroup.prototype.sendOffsetCommitRequest = function (commits, cb) {
...
+    return;
+  }
+
+  this.client.topicExists(topics, error => {
+    if (error) {
+      return cb(error);
+    }
+
+    async.waterfall([
```
[the remark this answers is not in the document]
Comment (LeiSun-1101) [reply lib/consumerGroup.js]: done

File: lib/consumerGroup.js [hunk 4]
```diff
@@ -611,6 +611,48 @@ ConsumerGroup.prototype.sendOffsetCommitRequest = function (commits, cb) {
   }
 };
 
+ConsumerGroup.prototype.addTopics = function (topics, cb) {
+  topics = Array.isArray(topics) ? topics : [topics];
+  let self = this;
```
[the remark this answers is not in the document]
Comment (LeiSun-1101) [reply lib/consumerGroup.js]: done

File: test/test.consumerGroup.js (hunk 2 above)
Comment (LeiSun-1101) [reply test/test.consumerGroup.js]: done

**Comment by LeiSun-1101** [comment]
Still need run the automatic tests many times. Any reason for that?

**Comment by hyperlink** [comment]
There are existing intermittent failures in the tests.

**Comment by hyperlink** [comment]
Thanks for your contribution!

**Pull request merged**